src/DynamoCoreWpf/Controls/StartPage.xaml.cs

```diff
@@ -111,9 +115,10 @@ public class StartPageViewModel : ViewModelBase
         ObservableCollection<StartPageListItem> recentFiles = null;
         internal readonly DynamoViewModel DynamoViewModel;
 
-        internal StartPageViewModel(DynamoViewModel dynamoViewModel)
+        internal StartPageViewModel(DynamoViewModel dynamoViewModel, bool isFirstRun)
         {
             this.DynamoViewModel = dynamoViewModel;
+            IsFirstRun = isFirstRun;
```

Comment:
Oops, didn't manage to catch this earlier, can we add a new data member `private readonly bool isFirstRun` and then here `this.isFirstRun = isFirstRun`? Then the property should read `public bool IsFirstRun { get { return isFirstRun; } }`.

This way we are enforcing that no one else (even in the class itself) can set the value of `isFirstRun`, avoiding possible mistakes in the class. Thanks!